src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -262,13 +264,10 @@ class GoogleExtensions(directoryDAO: DirectoryDAO, val accessPolicyDAO: AccessPo
       }
     } yield pet
 
-    createPet.unsafeToFuture() recoverWith {
-      case gjre: GoogleJsonResponseException if gjre.getDetails.getCode == StatusCodes.Conflict.intValue =>
-        googleIamDAO.findServiceAccount(project, petSaName).map { saOpt =>
-          val sa = saOpt.getOrElse(throw new WorkbenchException(s"Could not create pet service account for user [${user.email}]"))
-          PetServiceAccount(PetServiceAccountId(user.id, project), sa)
-        }
-    }
+    // googleServicesConfig.resourceNamePrefix is an environment specific variable passed in https://github.com/broadinstitute/firecloud-develop/blob/fade9286ff0aec8449121ed201ebc44c8a4d57dd/run-context/fiab/configs/sam/docker-compose.yaml.ctmpl#L24
+    // Use resourceNamePrefix to avoid collision between different fiab environments (we share same firestore for fiabs)
+    val lock = LockPath(CollectionName(s"${project.value}-createPet"), Document(user.id.value), 30 seconds)
```

Comment:
are there any limits on the length of collection or document names?

Reply:
seems `1,500 bytes` is the limit according to https://firebase.google.com/docs/firestore/quotas